entity-framework/core/miscellaneous/dbcontextpool.md

```diff
@@ -0,0 +1,38 @@
+---
+title: Connection pooling
+description: Connection pooling in Entity Framework Core
+author: rick-anderson
+ms.author: riande
+ms.date: 9/19/2020
+uid: core/miscellaneous/dbcontextpool
+---
+# Connection pooling
+
+`AddDbContextPool` enables `DbContext` pooling. Context pooling can increase throughput in high-scale scenarios such as web servers by reusing `DbContext` instances, rather than creating new instances for each request.
+
+The typical pattern in an ASP.NET Core app using EF Core involves registering a custom <xref:Microsoft.EntityFrameworkCore.DbContext> type into the [dependency injection](/aspnet/core/fundamentals/dependency-injection) container and obtaining instances of that type through constructor parameters in controllers or Razor Pages. Using constructor injection, a new instance of the `DbContext` is created for each request.
+
+<xref:Microsoft.Extensions.DependencyInjection.EntityFrameworkServiceCollectionExtensions.AddDbContextPool%2A> enables a pool of reusable `DbContext` instances. To use `DbContext` pooling, use the `AddDbContextPool` instead of `AddDbContext` during service registration:
+
+``` csharp
+services.AddDbContextPool<BloggingContext>(
+    options => options.UseSqlServer(connectionString));
+```
+
+When `AddDbContextPool` is used, at the time a `DbContext` instance is requested, EF first checks if there is an instance available in the pool. Once the request processing finalizes, any state on the instance is reset and the instance is itself returned to the pool.
+
+This is conceptually similar to how connection pooling operates in ADO.NET providers and has the advantage of saving some of the cost of initialization of the `DbContext` instance.
+
+The `poolSize` parameter of <xref:Microsoft.Extensions.DependencyInjection.EntityFrameworkServiceCollectionExtensions.AddDbContextPool%2A> sets the maximum number of instances retained by the pool. Once `poolSize` is exceeded, new `DbContext` instances are not cached and  EF falls back to the non-pooling behavior of creating instances on demand.
+
+## Limitations
```

Comment:
We've seen quite a few people using context pooling (and running into various issues) in cases where we don't think they actually gain anything from it. The last paragraph warns users about this, but consider making this a bit more visible bystart this section with a note instructing users to not do this gratuitously, but only enable it after profiling/benchmarking and seeing that context initialization is significant.